icu4c/source/common/dictbe.cpp

```diff
@@ -1053,13 +1064,18 @@ CjkBreakEngine::CjkBreakEngine(DictionaryMatcher *adoptDictionary, LanguageType
     // Korean dictionary only includes Hangul syllables
     fHangulWordSet.applyPattern(UnicodeString(u"[\\uac00-\\ud7a3]"), status);
     fHangulWordSet.compact();
+    fHiraganaWordSet.applyPattern(UnicodeString(u"[:Hiragana:]"), status);
```

Comment:
I think your fHiraganaWordSet do not need to be a member variable but just a on stack object in the constructor, after your call loadJapaneseParticleAndAuxVerbs you then create the hiraganaWordSet on the stack the same way and then you can call https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/classicu_1_1UnicodeSetIterator.html to get out the content and put that into fSkipSet. Since your fParticleSet and fHiraganaWordSet now are use for the same reason and there are no reason we have two different check, we should just have one HashSet named fSkipSet an load both the particle and verbs + the Hiragana into it and check that once only.